cluster-location-metadata/README.md

```diff
@@ -0,0 +1,343 @@
+
+Depending on the customer's needs, a cluster in Frankfurt am Main, Germany, could bear either of the following three location identifiers:
+
+- Country-level precision: `eu-de`
+- Country subdivision level precision: `eu-de-he`
+- City level precision: `eu-de-he-frankfurt`
+
+Component details:
+
+- `CONTINENT_CODE`: The mandatory first part is a two-letter code for the global region/continent. Available are:
+  - `af` for Africa
+  - `an` for Antarctica (yes, unlikely, but who knows)
+  - `as` for Asia
+  - `eu` for Europe
+  - `na` for North America
+  - `oc` for Oceania
+  - `sa` for South America
+- `COUNTRY_CODE` is the two letter [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) code of the country the cluster is located in.
+- `SUBDIVISION_CODE` is the code for a subdivision of the country, like it might makes sense for larger countries like the U. S. If CITY_NAME is specified, the SUBDIVISION_CODE must be given.
+- `CITY_NAME` is a concise form of the city name the cluster is hosted in, also optional.
+
+#### Setting the region label in metrics
+
+Currently the `region` label for on-premises installations always has the value `onprem`. In cloud installations, the label is set to the cloud region identifier.
+
+For both cloud and on-premises installations/clusters, we want the `region` label of metrics to be set according to the `topology.kubernetes.io/region` label of the cluster. For metrics regarding an application, the region label should represent the cluster the application is deployed to.
+
+For example, given a management cluster named `example-mc` in region `eu-de-hamburg` has an App resource to deploy an app in workload cluster `example-wc` that is located in region `eu-fr-pac-aix`, the application's metrics (if having a region label at all) should be labelled with `region="eu-fr-pac-aix"`.
+
+### Alternative solutions
+
+- Instead of `topology.kubernetes.io/region`, we could specify our own label in the `giantswarm.io` namespace. The latter would make sense in case we wanted to mark that there would be some logical difference between these labels.
+
+- Instead of strings, we could introduce a system that provides geo coordinates. (Since the cloud providers already work with string identifiers, this draft opts for a solution that extends this paradigm to on-prem.)
+
+### Implementation plan
+
+Team Honeybadger would assemble the default hierarchy, at least up to the second level (country), and partly up to the third level (subdivision code). We don't need global coverage with level three initially, as we can still extend our code table once new customers are onboarded or more locations are needed. The maintained code table should be applied as part of the cluster creation form(s), for quick location selection/entry.
```

Comment:
this is just for the on-prem clusters, right?

Reply:
Right, the default language code system is for on-prem.